byteguide/libs/fs.py

```diff
@@ -356,7 +356,7 @@ def save(self, metadata: t.Optional[t.Dict] = None) -> None:
         metadata_file = self._get_metadata_file()
 
         with open(metadata_file, "w", encoding="utf-8") as f:
-            json.dump(metadata, f, indent=4)
+            json.dump(metadata, f, indent=4, default=list)
```

Comment:
why default is `list`? can you explain the reason?

Reply:
Your comment @ninadmhatre  made me do some more digging and it was not the default=list that was the "real" fix. 

The real culprit was that this line of code uses {} instead of [] 

![image](https://github.com/ninadmhatre/byteguide/assets/54413402/1829fef7-7987-435d-9cee-66e1bc154014)

This caused the tags array which was uploaded for registering to get converted to a set with {}, see debug output below  

![image](https://github.com/ninadmhatre/byteguide/assets/54413402/225aa733-c24c-4c0c-a41a-fabf64d4e03b)
![image](https://github.com/ninadmhatre/byteguide/assets/54413402/7889033c-cce4-44ef-bccf-d366c9c19a74)

I will commit the "real" fix instead :)